exercises/smartPointers/problem1.cpp

```diff
@@ -0,0 +1,74 @@
+
+
+#include <iostream>
+#include <array>
+
+
+/* --------------------------------------------------------------------------------------------
+ * Unique ownership.
+ *
+ * Always use smart pointers instead of `new`. A frequent source of memory leaks is a function
+ * that terminates in an unexpected way.
+ *
+ * Tasks
+ * 1) Compile and run the code below. Notice that the final count is `1`,
+ *    showing that the instance of LargeObject has not been deallocated.
+ * 2) Modify `doStuff()` (only) so to use a `std::unique_ptr` instead of a raw pointer.
+ *    The final count should be `0`, and the memory leak solved.
+ * --------------------------------------------------------------------------------------------
+ */
+
+
+// The class LargeObject emulates a large object.
+// One should avoid to copy it, and rather use
+// a pointer to pass it around.
+
+struct LargeObject {
+
+    std::array<double, 100000> data ;
+
+    // So to check for some potential memory leak,
+    // we count the constructions and destructions
+    static std::size_t count ;
```

Comment:
```suggestion
    inline static std::size_t count = 0;
```

Reply:
(The same goes for the subsequent problems)

Reply:
Ok.